libopae/plugins/xfpga/metrics/afu_metrics.c

```diff
@@ -0,0 +1,275 @@
+	}
+
+	// Serach for AFU Metrics BBB DFH
+	while (feature_def.dfh.eol != 0 && feature_def.dfh.next_header_offset != 0) {
+
+		bbs_offset = feature_def.dfh.next_header_offset;
+
+		result = fpgaReadMMIO64(handle, 0, feature_def.dfh.next_header_offset, &(feature_def.dfh.csr));
+		if (result != FPGA_OK) {
+			FPGA_ERR("Invalid handle file descriptor");
+			result = FPGA_NOT_SUPPORTED;
+			return result;
+		}
+
+		if (feature_def.dfh.type == FEATURE_TYPE_BBB) {
+
+			result = fpgaReadMMIO64(handle, 0, bbs_offset +0x8, &(feature_def.guid[0]));
+			if (result != FPGA_OK) {
+				FPGA_ERR("Invalid handle file descriptor");
+				result = FPGA_NOT_SUPPORTED;
+				return result;
+			}
+
+			result = fpgaReadMMIO64(handle, 0, bbs_offset + 0x10, &(feature_def.guid[1]));
+			if (result != FPGA_OK) {
+				FPGA_ERR("Invalid handle file descriptor");
+				result = FPGA_NOT_SUPPORTED;
+				return result;
+			}
+
+			if (feature_def.guid[0] == METRICS_BBB_ID_L &&
+				feature_def.guid[1] == METRICS_BBB_ID_H) {
+				*offset = bbs_offset;
+				return FPGA_OK;
+			} else	{
+				FPGA_ERR(" Metrics BBB Not Found \n ");
+			}
+
+		}
+
+	}
+
+	FPGA_ERR("AFU Metrics BBB Not Found \n ");
+	return FPGA_NOT_FOUND;
+}
+
+
+fpga_result get_afu_metric_value(fpga_handle handle,
+				fpga_metric_vector *enum_vector,
+				uint64_t metric_num,
+				struct fpga_metric *fpga_metric)
+{
+	fpga_result result                           = FPGA_OK;
+	uint64_t index                               = 0;
+	struct metric_bbb_value metric_csr;
+	struct _fpga_enum_metric *_fpga_enum_metric  = NULL;
+	uint64_t num_enun_metrics                    = 0;
+
+	memset_s(&metric_csr, 0, sizeof(metric_csr));
+
+	if (handle == NULL ||
+		enum_vector == NULL ||
+		fpga_metric == NULL) {
+		FPGA_ERR("Invlaid Input Paramters");
+		return FPGA_INVALID_PARAM;
+	}
+
+	result = fpga_vector_total(enum_vector, &num_enun_metrics);
+	if (result != FPGA_OK) {
+		FPGA_ERR("Failed to get metric total");
+		return FPGA_NOT_FOUND;
+	}
+
+
+	for (index = 0; index < num_enun_metrics; index++) {
+
+		_fpga_enum_metric = (struct _fpga_enum_metric *)	fpga_vector_get(enum_vector, index);
+
+		if (metric_num == _fpga_enum_metric->metric_num) {
+
+			result = fpgaReadMMIO64(handle, 0, _fpga_enum_metric->mmio_offset, &metric_csr.csr);
+
+				fpga_metric->value.ivalue = metric_csr.value;
+
+		}
+
+	}
+
+	return result;
+}
+
+fpga_result add_afu_metrics_vector(fpga_metric_vector *vector,
+				  uint64_t *metric_id,
+				  uint64_t group_value,
+				  uint64_t metric_value,
+				  uint64_t metric_offset)
+{
+	fpga_result result                      = FPGA_OK;
+	struct metric_bbb_group group_csr;
+	struct metric_bbb_value metric_csr;
+	char group_name[SYSFS_PATH_MAX]         = { 0 };
+	char metric_name[SYSFS_PATH_MAX]        = { 0 };
+	char qualifier_name[SYSFS_PATH_MAX]     = { 0 };
+	char metric_units[SYSFS_PATH_MAX]       = { 0 };
+
+	memset_s(&group_csr, 0, sizeof(group_csr));
+	memset_s(&metric_csr, 0, sizeof(metric_csr));
```

Comment:
The signature of memset_s is..
```c
errno_t memset_s(void *dest, rsize_t len, uint8_t value);
```
This call isn't correct.  Please fix.

Reply:
fixed